commons/src/main/java/fr/free/nrw/commons/caching/CacheController.java

```diff
@@ -0,0 +1,94 @@
+package fr.free.nrw.commons.caching;
+
+import android.util.Log;
+
+import com.github.varunpant.quadtree.Point;
+import com.github.varunpant.quadtree.QuadTree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import fr.free.nrw.commons.upload.MwVolleyApi;
+
+public class CacheController {
+
+    private double x, y;
+    private QuadTree quadTree;
+    private Point[] pointsFound;
+    private double xMinus, xPlus, yMinus, yPlus;
+
+    private static final String TAG = CacheController.class.getName();
+
+    public CacheController() {
+        quadTree = new QuadTree(-180, -90, +180, +90);
+    }
+
+    public void setQtPoint(double decLongitude, double decLatitude) {
+        x = decLongitude;
+        y = decLatitude;
+        Log.d(TAG, "New QuadTree created");
+        Log.d(TAG, "X (longitude) value: " + x + ", Y (latitude) value: " + y);
+    }
+
+    public void cacheCategory() {
+        List<String> pointCatList = new ArrayList<String>();
+        if (MwVolleyApi.GpsCatExists.getGpsCatExists() == true) {
+             pointCatList.addAll(MwVolleyApi.getGpsCat());
+            Log.d(TAG, "Categories being cached: " + pointCatList);
+        } else {
+            Log.d(TAG, "No categories found, so no categories cached");
+        }
+        quadTree.set(x, y, pointCatList);
+    }
+
+    public List findCategory() {
+        //Convert decLatitude and decLongitude to a coordinate offset range
+        convertCoordRange();
+        pointsFound = quadTree.searchWithin(xMinus, yMinus, xPlus, yPlus);
+        List displayCatList = new ArrayList();
```

Comment:
Hmm.. I'm not sure what type parameters to use, as point.getValue() returns Objects. And using the Object type parameter seems a bit redundant?